### PR TITLE
Feature/46 pagination

### LINKED ISSUE #46: feat: Pagination
## Feature Description
Implement pagination for listing results, in order to allow users to navigate through pages of results
## Motivation
With this feature,data load and render times are reduced and provide the user with a more confortable experience when navigating through data
## Acceptance Criteria
- [ ] Criteria 1: users are able to navigate through pages of results, with a fixed amount of maximum elements

### PR DESCRIPTION
**Pull request by sebasruii**
# Description

Implemented logic for pagination in frontend. For now, I've only implemented the buttons and page navigation functionality, but I wouldn't merge this feature until Backend confirms the way pagination will be handled.

# UPDATE
I've implemented pagination for beneficiaries and interventions. The pagination elements are part of Material UI, so pnpm install is required prior to running the application. There's also a select input from which the user can select the number of elements displayed per page

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Screenshot(If applies):
image

### CODE REVIEW

**Comment by Ronald752** [comment]
I have reviewed your functionality, I understand that it can not be tested correctly because it lacks specifications on how the data will be treated in the backend, likewise I believe that when you specify should take into account these suggestions:
1.Verify that each data corresponds with the page number.
2.The number of pages, which I suppose will be a data that will send backend.
3.If it is possible to add a parameter to specify the order in which the data will be listed.

**Comment by sebasruii** [comment]
This PR will be moved to drafts, since the functionaity won't be completed until backend confirms the implications for the implementations. We'll be back! 👋🏻

**Comment by sebasruii** [comment]
This PR is ready for review. As of now, I've been deploying the backend implementend in the branch fix/288-count-documents-for-pagination

File: src/app/beneficiaries/fetch.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-		return beneficiaries.data
-	}
-    catch (error) {
+	try {
+		if (limit !== null) {
+			const beneficiaries = await axios.get(
+				`${BASEURL}/acat/patient?limit=${limit}&offset=${offset}`
+			)
+			return beneficiaries.data
+		} else {
```
Comment (TunixR) [inline src/app/beneficiaries/fetch.js]: shouldn't this take in account a possible offset null value and assign a default one?

**Review by TunixR — CHANGES_REQUESTED** [review]

File: src/app/beneficiaries/fetch.js (hunk 1 above)
Comment (sebasruii) [reply src/app/beneficiaries/fetch.js]: Refactored 👍🏻

**Review by TunixR — APPROVED** [review]
This looks good to me

**Pull request merged**